tests/graph/testObjectBoostGraphInterface.cpp

```diff
@@ -0,0 +1,625 @@
+    : myG1( g1 ),
+    graph_vertex_index( boost::get( boost::vertex_index_t(), g2 ) ),
+    graph_vertex_position( boost::get( vertex_position_t(), g2) ),
+    myIndexMap( indexMap )
+  {}
+
+  void operator()( const Vertex1& v1, const Vertex2& v2 ) const {
+    //  std::size_t idx = myIndexMap[ v1 ];
+    // Does not work !
+    // put( graph_vertex_index, v2, idx);
+    vertex_position pos;
+    pos.myP = v1;
+    //std::cout << "vertex " << idx << " at " << pos.myP << std::endl;
+    put( graph_vertex_position, v2, pos);
+  }
+};
+template <typename Graph1, typename Graph2>
+struct my_edge_copier {
+  my_edge_copier(const Graph1& , Graph2& )
+  {}
+  template <typename Edge1, typename Edge2>
+    void operator()(const Edge1& /*v1*/, Edge2& /*v2*/) const {
+      // does nothing
+    }
+};
+
+TEST_CASE_METHOD(Fixture_object_diamond_with_hole, "Copy graph", "[copy]" ){
+  GIVEN( "A diamond object with graph properties" ){
+    typedef FixtureObject Graph;
+    typedef boost::graph_traits<Graph>::vertex_descriptor vertex_descriptor; // ie Object::Vertex
+    typedef boost::graph_traits<Graph>::vertices_size_type vertices_size_type; // ie Object::Size
+    THEN( "Test copy_graph"){
+
+      // trace.beginBlock ( "Testing AdjacencyListGraph with copy_graph ..." );
+      using StdVertexIndexMap = std::map< vertex_descriptor, vertices_size_type > ;
+      StdVertexIndexMap vertexIndexMap;
+      boost::associative_property_map< StdVertexIndexMap > propVertexIndexMap( vertexIndexMap );
+      using BGraph = boost::adjacency_list< boost::vecS, boost::vecS, boost::undirectedS, VertexProperties > ;
+      BGraph bG;
+      boost::copy_graph( obj_fixture, bG,
+          boost::vertex_copy( my_vertex_copier<Graph,BGraph,StdVertexIndexMap>( obj_fixture, bG, vertexIndexMap ) )
+          .edge_copy( my_edge_copier<Graph,BGraph>( obj_fixture, bG ) )
+          .vertex_index_map( propVertexIndexMap )
+          );
+      using vertex_descriptor_2 = boost::graph_traits< BGraph >::vertex_descriptor ;
+      using vertex_iterator_2 = boost::graph_traits< BGraph >::vertex_iterator ;
+      using GraphVertexPositionMap = boost::property_map< BGraph, vertex_position_t>::type ;
+      GraphVertexPositionMap vertexPos( boost::get( vertex_position_t(), bG) );
+      for ( std::pair<vertex_iterator_2, vertex_iterator_2>
+          vp = boost::vertices( bG ); vp.first != vp.second; ++vp.first )
+      {
+        vertex_descriptor_2 v1 = *vp.first;
+        vertex_position pos = boost::get( vertexPos, v1 );
+        // trace.info() << "- " << v1 << " was at " << pos.myP << std::endl;
+      }
+      INFO("after copy: Boost graph has " << num_vertices( bG ) << " vertices.");
+      CHECK(boost::num_vertices( bG ) == boost::num_vertices( obj_fixture ));
+      // trace.endBlock();
+
+    }
+  }
+}
+
+
+// TEST_CASE_METHOD(Fixture_object_diamond_with_hole, "Wagner-Stoe min-cut and Boykov-Kolmogorov max-flow", "[mincut][maxflow]" ){
```

Comment:
I had... I have just found a bug in ObjectBoostGraphInterface.h null_vertex().
I switched to default constructor there:
 vertex_descriptor() instead of old vertex_descriptor( 0 )

These tests have been moved inside the section of breadth_fist_visit test.